cephbackupstorage/cephbackupstorage/cephagent.py

```
@@ -1072,6 +1149,540 @@ def cancel(self, req):
+        """Raised when a path escape (Zip-Slip) attempt is detected."""
+        pass
+
+    @staticmethod
+    def unzip_package_and_get_files_size(install_path):
+        # Zip-Slip pre-check: verify no path escapes before extracting
+        try:
+            listing = shell.call("tar -tf %s" % linux.shellquote(install_path))
+            for entry in listing.strip().splitlines():
+                entry = entry.strip()
+                if entry.startswith('/') or linux.contains_path_traversal(entry):
+                    raise CephAgent.PathEscapeError("path escape detected in archive entry: %s" % entry)
```

Ruff (0.15.4)

[warning] Abstract `raise` to an inner function

(TRY301)